### PR TITLE
duti: update to new darwin SDK pattern

### PR DESCRIPTION
**Pull request by n-hass**
Patches the configure steps to skip some unnecessary SDK checks, and nixfmt the package file.

Part of #354146.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by n-hass** [comment]
Result of `nixpkgs-review pr 357745` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>duti</li>
  </ul>
</details>

File: pkgs/os-specific/darwin/duti/default.nix [hunk 1]
```diff
@@ -10,15 +16,14 @@ stdenv.mkDerivation rec {
     sha256 = "1pg4i6ghpib2gy1sqpml7dbnhr1vbr43fs2pqkd09i4w3nmgpic9";
   };
 
-  nativeBuildInputs = [autoreconfHook];
-  buildInputs = [ApplicationServices];
-  configureFlags = [
-    "--with-macosx-sdk=/homeless-shelter"
+  nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = lib.optional stdenv.hostPlatform.isDarwin apple-sdk_11;
```
Comment (emilazy) [inline pkgs/os-specific/darwin/duti/default.nix:20]: Are you sure this is necessary? It does not look like it uses any macOS 11 APIs, so `buildInputs` should be able to be omitted entirely.

File: pkgs/os-specific/darwin/duti/default.nix [hunk 2]
```diff
@@ -10,15 +16,14 @@ stdenv.mkDerivation rec {
...
 
-    # needed to prevent duti from trying to guess our sdk
-    # NOTE: this is different than stdenv.hostPlatform.config!
-    "--host=x86_64-apple-darwin18"
-  ];
+  # needed to prevent duti from trying to guess our sdk
+  # NOTE: this is different than stdenv.hostPlatform.config!
+  configureFlags =
+    lib.optionals stdenv.hostPlatform.isAarch64 [ "--host=aarch64-apple-darwin18" ]
+    ++ lib.optionals stdenv.hostPlatform.isx86_64 [ "--host=x86_64-apple-darwin18" ];
```
Comment (emilazy) [inline pkgs/os-specific/darwin/duti/default.nix:26]: Looking at https://github.com/moretension/duti/blob/8b5b9a08702d5de6a5bce57fade593fca46e126e/aclocal.m4, we do actually still need to work around their outdated SDK checks, but we should do it a little differently.

I suggest:

```nix
  preConfigure = ''
    configureFlagsArray+=(
      "--with-macosx-sdk=$SDKROOT"
      "--with-macosx-deployment-target=$MACOSX_DEPLOYMENT_TARGET"
    )
  '';
```

This should also obviate the need for `--host`.

**Review by emilazy — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/darwin/duti/default.nix (hunk 2 above)
Comment (n-hass) [reply pkgs/os-specific/darwin/duti/default.nix]: Sure, changed. But I can’t see how to remove the host flag - you still end up with failure on newer systems as host will be set to an unhandled version (eg `darwin24` on latest macOS). Any suggestions, or are you happy with the latest commit?

Comment (emilazy) [reply pkgs/os-specific/darwin/duti/default.nix]: Hmm, that’s very unfortunate, it’s doing some strange and unnecessary stuff. I see two options:

* Pass `--host=${stdenv.hostPlatform.parsed.cpu.name}-apple-darwin20` and put `substituteInPlace Makefile.in --replace-fail '@macosx_arches@' ""` in a `postPatch` phase to work around its suboptimal architecture selection (might be able to get away without the latter part, not sure).

* Add a patch that comments out `DUTI_CHECK_SDK` and `DUTI_CHECK_DEPLOYMENT_TARGET` in `configure.ac` and replaces [all this](https://github.com/moretension/duti/blob/8b5b9a08702d5de6a5bce57fade593fca46e126e/Makefile.in#L17-L20) with just `OPTOPTS=	@OPTOPTS@`. Then we don’t need any configure flags at all and it should just work indefinitely. This is probably the cleaner approach.

**Review by emilazy — APPROVED** [review]
Looks good, thanks!

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357745`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>duti</li>
  </ul>
</details>

**Comment by emilazy** [comment]
Could you handle the backport?

**Pull request merged**